Vikta-API-testing-sample/src/test/java/com/test/api/payment/PaymentCardTest.java

```diff
@@ -0,0 +1,417 @@
+     *
+     * @param id paymentcard identifier
+     * @return PaymentCardDTO object
+     */
+    private PaymentCardDTO getPaymentCardById(long id) {
+        return given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .get(getTestEnvironment().getPaymentCardPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract().as(PaymentCardDTO.class);
+    }
+
+    /**
+     * Methods returns List of PaymentCards
+     *
+     * @return List<PaymentCardDTO>
+     */
+    private List<PaymentCardDTO> getPaymentCardList() {
+        return given().spec(defaultRequestSpec())
+                .when()
+                .log().all()
+                .get(getTestEnvironment().getPaymentCardListPath())
+                .then().
+                log().all()
+                .extract().jsonPath().getList(".", PaymentCardDTO.class);
+    }
+
+    private long getExistingUserId(){
+        return  (long) given().spec(defaultRequestSpec())
+                .when()
+                .get(getTestEnvironment().getUsersListPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", UserDTO.class)
+                .get(0).getId();
+    }
+
+    // here can't use getExistingUserId() method, because is not static and can't make it static because has other methods like getTestEnvironment()
+    private static Stream<Arguments> goodParametersForPost() {
+        long userId = 147;
+        return Stream.of(
+                Arguments.of(RandomStringUtils.randomAlphabetic(1), RandomStringUtils.randomNumeric(15), "123", "Jan", "2022-02-12", userId, "Case with min cardNickName length"),
+                Arguments.of(RandomStringUtils.randomAlphabetic(4096), RandomStringUtils.randomNumeric(15), "123", "Jan", "2022-02-12", userId, "Case with max cardNickName length"),
+                Arguments.of("Card Nick Name", RandomStringUtils.randomNumeric(14), "123", "Jan", "2022-02-12", userId, "Case with min cardNumber length"),
+                Arguments.of("Card Nick Name", RandomStringUtils.randomNumeric(16), "123", "Janusz", "2022-02-12", userId, "Case with min cardNumber length"),
+                Arguments.of("Card Nick Name", RandomStringUtils.randomNumeric(15), RandomStringUtils.randomNumeric(4), "Jan", "2022-02-12", userId, "Case with max cardCode length"),
+                Arguments.of("Card Nick Name 123", RandomStringUtils.randomNumeric(15), "123", RandomStringUtils.randomAlphabetic(1), "2022-02-12", userId, "Case with min ownerName length"),
+                Arguments.of("Card Nick 123 Name", RandomStringUtils.randomNumeric(15), "123", RandomStringUtils.randomAlphabetic(4096), "2022-02-12", userId, "Case with max ownerName length")
+        );
+    }
+
+    //same as above
+    // card code accepts negative numbers, shouldn't
```

Comment:
make sure to keep track of all weird cases. group the failing arguments at the bottom of the list, comment them out and write a reason above each of them